fe/fe-core/src/main/java/org/apache/doris/http/rest/TabletDiskDistributionAction.java

```diff
@@ -0,0 +1,192 @@
+        Map<Pair<Long, Integer>, Long> tabletToPathHash = new HashMap<Pair<Long, Integer>, Long>();
+        Map<Long, String> partitionToTable = new HashMap<Long, String>();
+        int tablet_total = 0;
+        List<Long> dbIds = Catalog.getCurrentCatalog().getDbIds();
+        for (long dbId : dbIds) {
+            Database db = Catalog.getCurrentCatalog().getDb(dbId);
+            if (db == null) {
+                continue;
+            }
+            db.readLock();
+            try {
+                List<Table> tables = db.getTables();
+                for (Table table : tables) {
+                    if (table == null) {
+                        continue;
+                    }
+                    if (table.getType() != TableType.OLAP) {
+                        continue;
+                    }
+                    String tableName = table.getName();
+                    OlapTable olapTable = (OlapTable) table;
+                    for (Partition partition : olapTable.getPartitions()) {
+                        if (partition == null) {
+                            continue;
+                        }
+                        long partitionId = partition.getId();
+                        partitionToTable.put(partitionId, tableName);
+                        List<Pair<Long, Integer>> tabletSchemaHash = Lists.newArrayList();
+                        for (MaterializedIndex index : partition.getMaterializedIndices(IndexExtState.ALL)) {
+                            for (Tablet tablet : index.getTablets()) {
+                                if (tablet == null) {
+                                    continue;
+                                }
+                                for (Replica replica : tablet.getReplicas()) {
+                                    if (replica == null) {
+                                        continue;
+                                    }
+                                    if (replica.getBackendId() == backend.getId()) {
+                                        Pair<Long, Integer> pair = Pair.create(tablet.getId(), olapTable.getSchemaHashByIndexId(index.getId()));
+                                        tabletSchemaHash.add(pair);
+                                        tabletToDataSize.put(pair, replica.getDataSize());
+                                        tabletToPathHash.put(pair, replica.getPathHash());
+                                        tablet_total++;
+                                    }
+                                }
+                            }
+                        }
+                        partitionToTablet.put(partitionId, tabletSchemaHash);
+                    }
+                }
+            } catch (Exception e) {
+                LOG.info(e);
+            } finally {
+                db.readUnlock();
+            }
+        }
+
+        JSONObject jsonObject_backend = new JSONObject();
```

Comment:
How about split this function to two parts? One for collecting info, the other for construct JSON?